lib/tasks/subscriber_list_audit.rake

```diff
@@ -0,0 +1,45 @@
+require "sidekiq/api"
+require "sitemap-parser"
+
+namespace :subscriber_list_audit do
+  desc "Create audit workers"
+  task :start, %i[batch_size limit] => :environment do |_t, args|
+    batch_size = (args[:batch_size] || "100").to_i
+    limit = args[:limit] ? args[:limit].to_i : -1
+
+    puts("Creating audit workers with batch size #{batch_size}, limit #{limit}")
+
+    base_sitemap = SitemapParser.new("#{Plek.website_root}/sitemap.xml")
+    sitemap_parts = base_sitemap.sitemap.at("sitemapindex").element_children.map { |element| element.at("loc").content }
+
+    sitemap_parts = limit == -1 ? sitemap_parts : sitemap_parts.take(1)
+
+    sitemap_parts.each do |sitemap_url|
+      sitemap = SitemapParser.new(sitemap_url)
+      urls = sitemap.to_a
+
+      urls = urls.take(limit) if limit != -1
+
+      puts("Read #{urls.count} URLs from sitemap part #{i}")
+
+      urls.each_slice(batch_size) do |batch|
+        SubscriberListAuditWorker.perform_async(batch)
+      end
+    end
+
+    puts("Batch workers created. Use rails subscriber_list_audit:queue_size to monitor queue size")
+  end
+
+  desc "Check audit worker queue size"
+  task queue_size: :environment do
+    audit_queue = Sidekiq::Queue.new(:subscriber_list_audit)
+    puts("#{audit_queue.size} jobs remaining to be processed")
+  end
+
+  desc "Show Subscriber Lists with no last audit"
+  task report: :environment do
+    SubscriberList.where(last_audited_at: nil) do |subscriber_list|
```

Comment:
Is this task supposed to be run as a one off? I'm just thinking that `last_audited_at` could get a value for a list if say the audit is run today when the list is in use, but still have a the old value 6 months from now whether the list is in use or not..

Reply:
Yeah, I think I was just trying to avoid for this run the horror show of dates in rake tasks :) but yes, if we run this more than once (which is likely for a while until we're clear that lists are getting tidied up), we need to be able to specify a cut-off, so I'll add an optional date.

Reply:
Okay, dates are in there now (defaults to today). To make things a bit easier to compare, I've also made the worker set the last_audited_at to the time when the audit was started, not time.zone.now. So all records that are marked good in a given audit will have the same audit time.